include/deal.II/sundials/ida.h

```diff
@@ -0,0 +1,653 @@
+     * @param ignore_algebraic_terms_for_errors Ignore algebraic terms for error computations
+     * @param ic_type Initial condition type
+     * @param reset_type Initial condition type after restart
+     * @param ic_alpha Initial condition Newton parameter
+     * @param ic_max_iter Initial condition Newton max iterations
+     * @param max_non_linear_iterations Maximum number of nonlinear iterations
+     * @param verbose Show output of time steps
+     * @param use_local_tolerances Use local tolerances
+     */
+    IDA(const MPI_Comm mpi_comm = MPI_COMM_WORLD,
+        const double &initial_time = 0.0,
+        const double &final_time = 1.0,
+        const double &initial_step_size = 1e-4,
+        const double &min_step_size = 1e-6,
+        const double &abs_tol = 1e-6,
+        const double &rel_tol = 1e-5,
+        const unsigned int &max_order = 5,
+        const double &output_period = .1,
+        const bool &ignore_algebraic_terms_for_errors = true,
+        const std::string &ic_type = "none",
+        const std::string &reset_type = "none",
+        const double &ic_alpha = .33,
+        const unsigned int &ic_max_iter = 5,
+        const unsigned int &max_non_linear_iterations = 10,
+        const bool &verbose = true,
+        const bool &use_local_tolerances = false);
+
+    /**
+     * House cleaning.
+     */
+    ~IDA();
+
+    /**
+     * Add all internal parameters to the given ParameterHandler object. When
+     * the paramaters are parsed from a file, the internal parameters are automatically
+     * updated.
+     *
+     * The following parameters are declared:
+     *
+     * @code
+     * set Absolute error tolerance                      = 0.000001
+     * set Final time                                    = 1.000000
+     * set Ignore algebraic terms for error computations = true
+     * set Initial condition Newton max iterations       = 5
+     * set Initial condition Newton parameter            = 0.330000
+     * set Initial condition type                        = use_y_diff
+     * set Initial condition type after restart          = use_y_dot
+     * set Initial step size                             = 0.000100
+     * set Initial time                                  = 0.000000
+     * set Maximum number of nonlinear iterations        = 10
+     * set Maximum order of BDF                          = 5
+     * set Min step size                                 = 0.000001
+     * set Relative error tolerance                      = 0.000010
+     * set Show output of time steps                     = true
+     * set Time units between each output                = 0.05
+     * set Use local tolerances                          = false
+     * @endcode
+     *
+     * These are one-to-one with the options you can pass at construction time.
+     *
+     * Those options are set as default value in the ParameterHandler object
+     * `prm`.
+     */
+    virtual void add_parameters(ParameterHandler &prm);
```

Comment:
an alternative is to move those into an auxiliary class `AdditionalParameters` which can declare and read things from ParameterHandler. Then you can easily extend the number of parameters later without breaking constructor interface. 

What I find more important is that you can then make all private members `const`.